### PR TITLE
MINOR: Fix Kafka Streams JavaDocs with regard to new StreamJoined class

### PR DESCRIPTION
**Pull request by mjsax**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @bbejeck @cadonna

File: streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala [hunk 1]
```diff
@@ -466,7 +466,9 @@ class KStream[K, V](val inner: KStreamJ[K, V]) {
    * @return a [[KStream]] that contains join-records for each key and values computed by the given `joiner`,
    * one for each matched record-pair with the same key
    * @see `org.apache.kafka.streams.kstream.KStream#join`
+   * @deprecated since 2.4. Use [[KStream#koin(KStream, ValueJoiner, JoinWindows, StreamJoined)]] instead.
```
Comment (mjsax) [inline streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala]: I hope this is the correct markup?

Comment (bbejeck) [reply streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala]: I think so

Comment (bbejeck) [inline streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala]: typo: `KStream#koin` -> `KStream#join`

**Review by bbejeck — APPROVED** [review]
Thanks for the PR @mjsax! I have only one minor comment, otherwise LGTM.

**Review by cadonna — APPROVED** [review]
LGTM, Thanks @mjsax

**Comment by bbejeck** [comment]
failures unrelated

**Comment by bbejeck** [comment]
Merged #7627 into trunk

**Pull request merged**